src/main/kotlin/no/nav/familie/ef/sak/beregning/BeregningBarnetilsynService.kt

```diff
@@ -0,0 +1,112 @@
+package no.nav.familie.ef.sak.beregning
+
+import no.nav.familie.ef.sak.beregning.barnetilsyndto.BeløpsperiodeBarnetilsynDto
+import no.nav.familie.ef.sak.beregning.barnetilsyndto.BeregningsgrunnlagBarnetilsynDto
+import no.nav.familie.ef.sak.felles.dto.Periode
+import org.springframework.stereotype.Service
+import java.math.BigDecimal
+
+@Service
+class BeregningBarnetilsynService {
+
+    fun beregnYtelseBarnetilsyn(utgiftsperioder: List<UtgiftsperiodeDto>,
+                                kontantstøttePerioder: List<KontantstøttePeriodeDto>,
+                                tilleggsstønadsperioder: List<TilleggsstønadPeriodeDto>): List<BeløpsperiodeBarnetilsynDto> {
+
+        val barnetilsynMåneder = utgiftsperioder.map {
+            it.split()
+        }.flatMap {
+            it.map { utgiftsMåned ->
+                utgiftsMåned.tilBeløpsperiodeBarnetilsynDto(kontantstøttePerioder, tilleggsstønadsperioder)
+            }
+        }
+        return barnetilsynMåneder.mergeSammenhengendePerioder()
+    }
+}
+
+/**
+ * Del opp utgiftsperioder i atomiske deler (mnd).
+ * Eksempel: 1stk UtgiftsperiodeDto fra januar til mars deles opp i 3:
+ * listOf(UtgiftsMåned(jan), UtgiftsMåned(feb), UtgiftsMåned(mars))
+ */
+fun UtgiftsperiodeDto.split(): List<UtgiftsMåned> {
+    val perioder = mutableListOf<UtgiftsMåned>()
+    var måned = this.årMånedFra
+    while (måned.isBefore(this.årMånedTil) || måned.equals(this.årMånedTil)) {
```

Comment:
Her gir intellij warning
```suggestion
    while (måned.isBefore(this.årMånedTil) || måned == this.årMånedTil) {
```

Reply:
Men så kan man også bruke
```suggestion
    while (måned <= this.årMånedTil) {
```

Reply:
likte: while (måned <= this.årMånedTil) :)